YarnSpinner/Value.cs

```diff
@@ -0,0 +1,334 @@
+
+using System;
+
+namespace Yarn
+{
+	// A value from inside Yarn.
+	public class Value : IComparable, IComparable<Value> {
+        public static readonly Value DEFAULT = new Value();
+        
+		public enum Type {
+            Undefined = 0, // we don't know yet or don't have a value
+			Number,  // a constant number
+			String,  // a string
+			Bool,    // a boolean value
+			Variable, // the name of a variable; will be expanded at runtime
+			Null,    // the null value
+		}
+
+		public Value.Type type { get; internal set; }
+        
+        // this is primarily a debugging property, allowing
+        // for testing between actual and perceived type
+        internal object rawBacking { get; set; }
```

Comment:
Do we really need to have this? It means that we're storing data twice, and that makes me nervous. 

If we need a stronger way to ensure that the type that the Value thinks it is is the same as the value that was applied, adding logic to the underlying values (`numberValue`, `stringValue`, etc) that throws an exception if the wrong underlying value was accessed (eg if `numberValue` is accessed, but the Value is a bool, then throw an exception) might be a better idea. What do you think?